server/src/main/java/org/opensearch/common/io/stream/ByteBufferStreamInput.java

```diff
@@ -147,7 +147,7 @@ public int available() throws IOException {
     }
 
     @Override
-    protected void ensureCanReadBytes(int length) throws EOFException {
+    public void ensureCanReadBytes(int length) throws EOFException {
```

Comment:
I'm probably missing something here but these are all subclasses of streamInput, curious why the visibility was changed in the abstract?

Reply:
Aha! The visibility had to be raised in the other PR when I was holding onto `BaseStreamInput`. Since I eliminated that here, you're right, I can go back to protected. Nice catch!